### DISCUSSION TITLE
GitHub releases for canary versions?

### DISCUSSION THREAD

**Discussion by amannn** [ghiscoding/lerna-lite#874] [Q&A] [answered]
Hey, first of all a big thank you to @ghiscoding! I've been using Lerna for a longer time on the `next-intl` repo, but as part of updating the repo infrastructure I'm now considering moving to `lerna-lite`. The first impression is absolutely great!

I'm currently investigating setting up [a canary release channel](https://github.com/amannn/next-intl/pull/1118). Canary versions are already correctly published to npm under the `canary` tag, but I noticed that GitHub releases aren't published. Is this intended?

I'd be curious if there's a way how I could create GitHub releases for canaries too (like [Next.js does it](https://github.com/vercel/next.js/releases/tag/v15.0.0-canary.27)).

Any help would be appreciated, thanks!

---

As a side note, I was a bit surprised that multiple canaries can't be published from the same branch by default. I was able to find [a workaround](https://github.com/amannn/next-intl/pull/1118/commits/dc8fb0d9a83bfe2cfb8c39ebe295098621796271) though. I'm wondering if `lerna-lite` could provide better support for this?

**EDIT:** Ah no, seems like a different version is referenced than published.

![Screenshot 2024-06-12 at 19 10 07](https://github.com/lerna-lite/lerna-lite/assets/4038316/23e0d012-2a13-42c4-8694-caf4aaa19810)

![Screenshot 2024-06-12 at 19 10 17](https://github.com/lerna-lite/lerna-lite/assets/4038316/2481f1b5-d62f-4f54-8aa6-ef458c8cd20a)

**Comment by ghiscoding** [comment]
> I'm currently investigating setting up [a canary release channel](https://github.com/amannn/next-intl/pull/1118). Canary versions are already correctly published to npm under the `canary` tag, but I noticed that GitHub releases aren't published. Is this intended?

Do you mean it's not creating a GitHub Release on the GitHub website? I actually personally never even tried the canary feature, so apart from the available unit tests, I don't know how and what it produces since that all came from the original Lerna. I created Lerna-Lite fork when Lerna was out of support (before Nrwl took over Lerna), so for the most part it's all Lerna's original code. I typically tell the users to first search in Lerna's list of issue to see if anything comes up similar to your issue. 

Related to your problem, these Lerna issues might be relevent
- https://github.com/lerna/lerna/issues/2778  - this seems related to your problem with `-` and `+` in the same version
- https://github.com/lerna/lerna/issues/2622

So in your case I would really just go over Lerna and read any related issue and comments because I have no idea what is causing this problem since like I said, I have never used the canary releases myself (I typically just use alpha/beta tags). If canary aren't being pushed to GitHub website, that could probably be investigated though

**Reply by amannn** [reply]
Thanks for the quick reply!

Yeah I think the canary feature is somewhat buggy. I've now switched to using `--preid canary --dist-tag canary`, that seems to work for me.

**Reply by ghiscoding** [reply]
side note, since you're using GitHub Action to publish, you could also enable NPM `--provenance`, I enabled it in all my projects and I wrote an article on how to do it with Lerna-Lite

https://dev.to/ghiscoding/how-to-publish-on-npm-with-provenance-using-lerna-lite-3cjf

**Reply by amannn** [reply]
[I do have that already](https://github.com/amannn/next-intl/blob/99e069a1d7f5ded0b2d91ed574968d08f3365835/.github/workflows/release.yml#L31), thanks for the hint though! :)

**Comment by ghiscoding** [comment, marked as answer]
@amannn  Side note and now that I'm revisiting old discussions, I created a new PR #999 to change `+{SHA}` to `.{SHA}` and that should (hopefully) resolve the issues with Canary releases (there's more info in the PR as to what this fixes). I'm pretty sure that the issue you had was caused by that (npm disregard everything set after the + symbol because I think they changed it to be a reserved char, a dot would work though)

However, please note that it's a breaking change & it's scheduled for the upcoming v4 major release which will come real soon

**Reply by amannn** [reply]
That sounds like a helpful change! I'm currently using a [workaround](https://github.com/amannn/next-intl/blob/1e74950280453dedd2fda62734a00ae50353d47f/.github/workflows/prerelease-canary.yml#L26-L28) that has served me well. Maybe I'll reevaluate when you've released the v4 major! Thank you for your work on `lerna-lite`!

**Reply by ghiscoding** [reply]
this is now shipped, see the new major [v4.0](https://github.com/lerna-lite/lerna-lite/releases/tag/v4.0.0) release
